### ISSUE TITLE
payload is set but no content-type header specified

### ISSUE THREAD

**Issue by JAStanton** [getsentry/react-native-sentry#184]
While running Sentry on Android for the first time I got an error saying I needed to set content-type for header. "payload is set but no content-type header specified".

I am on:
```
react-native 0.44.3
react-native-sentry 0.15.1
```

It looks like it was supposed to be fixed here: https://github.com/getsentry/raven-js/pull/432 2 years ago. The Raven-js version that is ultimately being pulled in from react-native-sentry is version `3.17.0` which appears to be the latest. Looking at their [source](https://github.com/getsentry/raven-js/blob/3.17.0/dist/raven.js#L1769-L1770) it doesn't look like we're setting request header like in the [fix pr](https://github.com/getsentry/raven-js/pull/432/commits/1d37716181592aa737b6396a9fe08937259735c5).

Is there a different way I should go about setting this? Is there a way to configure react-native or raven differently to make this all compatible?

**Comment by HazAT** [comment]
Hey,
Sorry, I am bit confused.

Is this just happening on Android? 
Can you paste your logs here?
What does your configuration look like? Also, did you deactivate the native integration?

Adn wow do CocoaPods come into play?

**Comment by JAStanton** [comment]
Ah I guess pods are irrelevant seeing as it's an android build. My apologies. 

This is happening on Android. The error was `payload is set but no content-type header specified`, And yes I disabled native side. Something I realized was that the native side wasn't fully integrated at this time. That said this is running client side / raven-js only. When I manually changed raven-js in node_modules to include headers like the fix from 2 years ago it does it work. Obviously I'm missing something since that's been removed from some time.


Gimme a bit to get a full stack trace

**Comment by JAStanton** [comment]
Note: **I was able to fix this** without hacks by setting `disableNativeIntegration` to `false`. 

However it's important to note that this configure will fail on react-native:
```
deactivateStacktraceMerging: false
logLevel: SentryLog.Verbose
disableNativeIntegration: true
ignoreModulesExclude: []
ignoreModulesInclude: ["RNSentry"]
```

```
console.error: "Raven transport failed to send: ", {"request":{"UNSENT":0,"OPENED":1,"HEADERS_RECEIVED":2,"LOADING":3,"DONE":4,"readyState":4,"status":0,"timeout":0,"withCredentials":true,"upload":{},"_aborted":false,"_hasError":true,"_method":"POST","_response":"Payload is set but no content-type header specified","_url":"https://sentry.io/api/200816/store/?sentry_version=7&sentry_client=raven-js%2F3.17.0&sentry_key=XXX&sentry_secret=XXXX","_timedOut":false,"_trackingName":"unknown","_incrementalEvents":false,"_requestId":null,"_headers":{},"_responseType":"","_sent":true,"_lowerCaseResponseHeaders":{},"_subscriptions":[]}}
_logDebug
    raven.js:1712:16
failure
    raven.js:1643:31
onreadystatechange
    raven.js:1680:33
wrapped
    raven.js:321:34
dispatchEvent
    event-target.js:172:43
setReadyState
    XMLHttpRequest.js:538:23
__didCompleteResponse
    XMLHttpRequest.js:381:25
<unknown>
    XMLHttpRequest.js:485:16
emit
    EventEmitter.js:182:12
__callFunction
    MessageQueue.js:250:47
<unknown>
    MessageQueue.js:101:26
__guard
    MessageQueue.js:218:6
callFunctionReturnFlushedQueue
    MessageQueue.js:100:17
```

**Comment by HazAT** [comment]
Hey, this should be fixed in `0.17.0`

**Comment by xwartz** [comment]
Why it can't compatible with Android?
